components/procore/sources/common.js

```diff
@@ -0,0 +1,70 @@
+const procore = require("../procore.app.js");
+
+module.exports = {
+  dedupe: "unique",
+  props: {
+    procore,
+    db: "$.service.db",
+    http: "$.interface.http",
+    company: { propDefinition: [procore, "company"] },
+    project: {
+      propDefinition: [procore, "project", (c) => ({ company: c.company })],
+    },
+  },
+  methods: {
+    getEventTypes() {
+      return ["create", "update", "delete"];
+    },
```

Comment:
Minor: even better would be to use the `eventTypes` array in `procore.app.js`:
```js
module.exports = {
  methods: {
    getEventTypes() {
      return procore.eventTypes.map(({ value }) => value);
    };
  },
};
```

This way, any change we do in `procore.eventTypes` will be automatically reflected by this component.